pkg/kubectl/cmd/util/factory.go

```diff
@@ -303,6 +307,34 @@ func (f *Factory) BindFlags(flags *pflag.FlagSet) {
 	flags.SetNormalizeFunc(util.WordSepNormalizeFunc)
 }
 
+func (f *Factory) AddDeprecatedFlagShorthands(shorthand string) error {
```

Comment:
I think that's a good idea to move this to pflag. Doing it "right" will take a bit more work. Obviously right now we only track if the short and long are both deprecated. So we'll need a way to say that only the short is deprecated. We really should start hiding the short from help text and documentation generation. I don't think `Set()` is actually the place you'll be able to find the short flag. I think it only handles long flags. You'll need to look at `origArg[0]` in `setFlag()`.

Also `CheckDeprecatedFlagShorthands()` isn't right...  It is legal (although silly) to do something like `-wt '{ name }'` (or something like there) where you have multiple shorthands after one `-`